src/structs/jpeg_header.rs

```diff
@@ -434,30 +477,37 @@ impl JPegHeader {
             }
         }
 
-        self.mcuv = (1.0 * self.img_height as f64 / (8.0 * self.sfhm as f64)).ceil() as i32;
-        self.mcuh = (1.0 * self.img_width as f64 / (8.0 * self.sfvm as f64)).ceil() as i32;
-        self.mcuc = self.mcuv * self.mcuh;
+        self.mcuv = NonZeroU32::new(
+            (1.0 * self.img_height as f64 / (8.0 * self.sfhm as f64)).ceil() as u32,
+        )
+        .ok_or_else(|| LeptonError::new(ExitCode::UnsupportedJpeg, "mcuv is zero"))?;
+
+        self.mcuh =
+            NonZeroU32::new((1.0 * self.img_width as f64 / (8.0 * self.sfvm as f64)).ceil() as u32)
+                .ok_or_else(|| LeptonError::new(ExitCode::UnsupportedJpeg, "mcuh is zero"))?;
```

Comment:
Why not make use of integer arithmetics? And I have an impression that these checks are superfluous: checks of nonzero `img_height` and `img_width` are enough.

Reply:
Yeah I guess that's just a round up